bigbio/biodatasets/medical_data/medical_data.py

```diff
@@ -0,0 +1,146 @@
+# limitations under the License.
+
+import os
+from typing import List, Tuple, Dict
+import pandas as pd
+import datasets
+from bigbio.utils import schemas
+from bigbio.utils.configs import BigBioConfig
+from bigbio.utils.constants import Tasks
+
+_CITATION = """\
+@misc{ask9medicaldata,
+  author    = {Khan, Arbaaz},
+  title     = {Sentiment Analysis for Medical Drugs},
+  year      = {2019},
+  url       = {https://www.kaggle.com/datasets/arbazkhan971/analyticvidhyadatasetsentiment},
+}
+"""
+
+_DATASETNAME = "medical_data"
+
+_DESCRIPTION = """\
+    This dataset is designed to do multiclass classification on medical drugs
+"""
+
+_HOMEPAGE = ""
+
+_LICENSE = ""
+
+_URLS = {}
+
+_SUPPORTED_TASKS = [Tasks.TEXTUAL_ENTAILMENT]
+
+
+_SOURCE_VERSION = "1.0.0"
+
+_BIGBIO_VERSION = "1.0.0"
+
+
+class MedicaldataDatatset(datasets.GeneratorBasedBuilder):
+    """This dataset contains comments about patients and the sentiment in those comments about a specific drug that's mentioned.
+    1 - Negative sentiment
+    2 - Positive sentiment
+    0 - Neutral"""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    BIGBIO_VERSION = datasets.Version(_BIGBIO_VERSION)
+
+    BUILDER_CONFIGS = [
+        BigBioConfig(
+            name=f"{_DATASETNAME}_source",
+            version=SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=f"{_DATASETNAME}",
+        ),
+        BigBioConfig(
+            name=f"{_DATASETNAME}_bigbio_te",
+            version=BIGBIO_VERSION,
+            description=f"{_DATASETNAME} BigBio schema",
+            schema="bigbio_te",
+            subset_id=f"{_DATASETNAME}",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+
+        if self.config.schema == "source":
+
+            features = datasets.Features(
+                {
+                    "hash": datasets.Value("string"),
+                    "text": datasets.Value("string"),
+                    "drug_name": datasets.Value("string"),
+                    "sentiment": datasets.Value("uint32"),
+                }
+            )
+
+        elif self.config.schema == "bigbio_te":
+            features = schemas.entailment_features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        if self.config.data_dir is None:
+            raise ValueError(
+                "This is a local dataset. Please pass the data_dir kwarg to load_dataset."
+            )
+        else:
+            data_dir = self.config.data_dir
+
+        return [
+            datasets.SplitGenerator(
```

Comment:
@shamikbose thanks for the quick turnaround on this dataset! When I downloaded this dataset from Kaggle, there was also a test set (`test_tOlRoBf.csv`). Could you please add that split as well?

Reply:
The test set does not have labels. Here's an example entry

```unique_hash,text,drug
9e9a8166b84114aca147bf409f6f956635034c08,"256 (previously stable on natalizumab), with 55% switching to fingolimod",fingolimod
e747e6822c867571afe7b907b51f0f2ca67b0e1a,On fingolimod and have been since December 2015; the only way I can describe it any better is like feeling drunk but without having had a drink!!!,fingolimod
50b6d851bcff4f35afe354937949e9948975adf7,"Apparently it's shingles! :-/ I do have a few red spots just below my left breast which appeared a few days ago. Wasn't sure if they were connected or not, but showed them to the GP and he immediately said shingles. Apparently 'cos I left it a few days it was too late to prescribe me anything for shingles. The GP prescribed me some useless amitryptiline, which I probably won't take because even a low dose makes me unbearably groggy. Still not entirely happy, just 'cos the stomach pain is so bad and I've had a lot of bloating and stuff. No fever as far as I know. Anyone here had any experience with shingles? Seems like I can expect 3-4 more weeks of this ;_; Dx Crohn's in June 2000. (Yay  ) Tried: 5-ASAs, azathioprine, 6MP, Remicade, methotrexate, Humira, diets. 1st surgery 20/2/13 - subtotal colectomy with end ileostomy. 2nd surgery 10/7/15 - ileorectal anastomosis. Stoma reversed and ileum connected to the rectum. Current status: Chronic flare. Do I have any other kind? Current meds: 50mg 6MP; Entyvio (started 3/11/16)",humira
```

Reply:
agree, we should add all splits. it's not uncommon for the test set to be missing some values. @hakunanatasha recently added some flags in the test suite for this https://github.com/bigscience-workshop/biomedical/blob/master/tests/test_bigbio.py#L824

Reply:
@ruisi-su @galtay Cool, I can use the split parameter. What should the default values be? It's an uint32 field. It can't be 0 since that's a valid value in the train set

Reply:
@ruisi-su @galtay In the latest commit, changed the sentiment to a `string` type. For missing values in the test split, it's set to `None`

Reply:
awesome. That looks good to me. `None` is fine as it is passing the unit tests.

Reply:
@ruisi-su Does this need any other changes or can it be closed?